lib/overcommit/hook/pre_push/protected_branches.rb

```diff
@@ -13,6 +14,15 @@ def run
 
     private
 
+    def log_pushed_refs
+      log_msg = "\nPushed refs:\n\t#{pushed_refs_str}"
+      Overcommit::Utils.log.debug(log_msg)
```

Comment:
Makes sense. We'd need to gate the run in a `begin...ensure` block so that log messages are printed even if the hook run throws an exception.